src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -118,6 +180,73 @@ class Select<T> extends React.Component<Props<T>> {
 
     handleOptionListClose = this.closeOptionList;
 
+    @computed get highestButtonIndex() {
+        const max = Math.max(...Array.from(this.buttonRefs.keys()));
+        return max;
+    }
+
+    @action requestFocus = (focusedElementIndex: number) => {
+        if (focusedElementIndex < 0 || focusedElementIndex > this.highestButtonIndex) {
+            return;
+        }
+
+        if (this.buttonRefs.has(focusedElementIndex)) {
+            this.focusedElementIndex = focusedElementIndex;
+            const ref = this.buttonRefs.get(focusedElementIndex);
+
+            if (ref) {
+                ref.focus();
+            }
+        }
+    };
+
+    handleRequestFocus = (focusedElementIndex: number) => () => {
+        this.requestFocus(focusedElementIndex);
+    };
+
+    @action handleKeyDown = (event: KeyboardEvent) => {
+        if (!this.open) {
+            return;
+        }
+
+        event.preventDefault();
+
+        if (event.key === 'Escape') {
+            this.closeOptionList();
+
+            return;
+        }
+
+        let focusedElementIndex = this.focusedElementIndex;
+
+        if (event.key === 'ArrowUp') {
+            do {
+                --focusedElementIndex;
+            } while (!this.buttonRefs.has(focusedElementIndex) && focusedElementIndex > 0);
+        }
+
+        if (event.key === 'ArrowDown') {
+            do {
+                ++focusedElementIndex;
+            } while (!this.buttonRefs.has(focusedElementIndex) && focusedElementIndex <= this.highestButtonIndex);
+        }
+
+        if (focusedElementIndex !== this.focusedElementIndex) {
+            this.requestFocus(focusedElementIndex);
+        }
+
+        this.clearSearchText();
```

Comment:
doesnt this always reset the search term? so the searchterm can never be longer than 1 character? 🤔